File: sample/CStyleSamplesCS/Samples/CaptureAVI.cs
/*
 * (C) 2008-2013 Schima
 * This code is licenced under the LGPL.
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenCvSharp;
using SampleBase;

namespace CStyleSamplesCS
{
    /// <summary>
    /// AVI capture
    /// </summary>
    class CaptureAVI
    {
        public CaptureAVI()
        {
            using (CvCapture cap = CvCapture.FromFile(FilePath.Movie.Hara)) 
            using (CvWindow w = new CvWindow("SampleCapture"))
            {
                int interval = (int)(1000 / cap.Fps);
                while (true)
                {
                    IplImage image = cap.QueryFrame();
                    if (image == null)
                    {
                        break;
                    }
                    w.Image = image;
                    if (Cv.WaitKey(interval) > 0)
                    {
                        break;
                    }
                }
            }
        }
    }
}

File: sample/CStyleSamplesVB/Samples/CaptureAVI.vb
'
' * (C) 2008-2013 Schima
' * This code is licenced under the LGPL.
' 

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' AVIファイルのキャプチャ
''' </summary>
Friend Module CaptureAVI
    Public Sub Start()
        Using cap As CvCapture = CvCapture.FromFile(FilePath.Movie.Hara)
            Using w As New CvWindow("SampleCapture")
                Dim interval As Integer = CInt(Math.Truncate(1000 / cap.Fps))
                Do
                    Dim image As IplImage = cap.QueryFrame()
                    If image Is Nothing Then
                        Exit Do
                    End If
                    w.Image = image
                    If Cv.WaitKey(interval) > 0 Then
                        Exit Do
                    End If
                Loop
            End Using
        End Using
    End Sub
End Module
' End Namespace
